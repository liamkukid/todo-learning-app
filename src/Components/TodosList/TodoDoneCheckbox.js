import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';

import { todoDone } from '../../features/todosSlice';

export default function TodoDoneCheckbox({ todo }) {
  const dispatch = useDispatch();
  const handleChange = e => {
    const { value, checked } = e.target;
    const payload = { id: value, done: checked };
    dispatch(todoDone(payload));
  };

  return (
    <input
      type="checkbox"
      value={todo.id}
      checked={todo.done}
      onChange={handleChange}
    />
  );
}

TodoDoneCheckbox.propTypes = {
  todo: PropTypes.shape({
    title: PropTypes.string,
    done: PropTypes.bool,
    id: PropTypes.string,
  }).isRequired,
};
