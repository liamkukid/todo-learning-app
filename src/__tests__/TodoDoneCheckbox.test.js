import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import todosReducer from '../features/todosSlice';
import TodoDoneCheckbox from '../Components/TodosList/TodoDoneCheckbox';

describe('TodoDoneCheckbox', () => {
  let store;
  const todo = {
    title: 'todo 1',
    done: false,
    id: '14e520cd-63f2-485e-9b3b-072ce068554e',
  };

  beforeEach(() => {
    store = configureStore({
      reducer: {
        todos: todosReducer,
      },
      preloadedState: {
        todos: [todo],
      },
    });
  });

  afterEach(() => {
    store = null;
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <TodoDoneCheckbox todo={todo} />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });

  it('change the state of todo done=true', () => {
    render(
      <Provider store={store}>
        <TodoDoneCheckbox todo={todo} />
      </Provider>
    );
    const linkElementButton = screen.getByTestId('todo-checkbox');

    fireEvent.click(linkElementButton);

    const state = store.getState();
    expect(state.todos[0].done).toBeTruthy();
  });
});
