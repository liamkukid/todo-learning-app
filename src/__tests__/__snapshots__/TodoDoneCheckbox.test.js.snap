// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoDoneCheckbox renders correctly 1`] = `
<input
  checked={false}
  data-testid="todo-checkbox"
  onChange={[Function]}
  type="checkbox"
  value="14e520cd-63f2-485e-9b3b-072ce068554e"
/>
`;
